capstone/src/main/java/com/google/univiz/CollegeDataApi.java

```diff
@@ -0,0 +1,13 @@
+package com.google.univiz;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * CollegeDataApi is an interface for the CollegeDataApiImpl class that interacts with external APIs
+ * to send REST requests.
+ */
+public interface CollegeDataApi {
+
+  List<CollegeData> getCollegesById(Set<CollegeId> ids);
```

Comment:
I think there's an impedance mismatch here between the `Set` of ids and the returned `List`. As a reader of this signature, it's unclear how the results will be ordered. Do you want to change the returned `List` to be a `Set`, as well?

Reply:
i dont think so. ill change it to a `List`

Reply:
I actually liked the set input better because it clearly communicates that this API does not like to get duplicate IDs. But i also agree that a set input should be followed by a set output. But returning a list of better than set for output because we like to have stuff returned in some deterministic order. Otherwise college names could keep jumping around every time the user refreshes the page. No action item for now. Let's keep both as a list.

Reply:
ok. we can talk about this at standup.